src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,150 @@
+        }
+    }
+
+    public void deleteTask(int index) {
+        try {
+            Task deletedTask = tasks.get(index);
+            tasks.remove(index);
+            storage.updateFile(tasks);
+            ui.toScreen("Ok, I've deleted the following task: ",
+                    "    " + deletedTask.toString());
+        } catch (IndexOutOfBoundsException e) {
+            ui.toScreen("Duke.Task could not be deleted.", "Please input valid task index.");
+        }
+    }
+
+    public void addTodo(String desc) {
+        try {
+            if (desc.isEmpty()) {
+                throw new DukeException();
+            }
+            Task temp = new Todo(desc);
+            tasks.add(temp);
+            storage.writeTask(temp);
+            ui.toScreen("Ok! A new task has been added:",
+                    "    " + temp.toString(),
+                    "You now have " + tasks.size() + " task(s) in total.");
+        } catch (DukeException de) {
+            ui.toScreen("Sorry, the Duke.Todo task could not be added.",
+                    "Please include the description for this task.", desc);
+        }
+    }
+
+    public void addDeadline(String[] req) {
+        try {
+            if (req.length != 2) {
+                throw new DukeException();
+            }
+            String desc = req[0];
+            String deadline = req[1];
+            Task temp = new Deadline(desc, deadline);
+            tasks.add(temp);
+            storage.writeTask(temp);
+            ui.toScreen("Ok! A new task has been added:",
+                    "    " + temp.toString(),
+                    "You now have " + tasks.size() + " task(s) in total.");
+        } catch (DukeException de) {
+            ui.toScreen("Sorry, the Duke.Deadline task could not be added.",
+                    "Please include the description and deadline for this task with /by.",
+                    "(Date and time format: dd/MM/yyyy HHmm)");
+        }
+    }
+
+    public void addEvent(String[] req) {
+        try {
+            if (req.length != 2) {
+                throw new DukeException("oops");
+            }
+            String desc = req[0];
+            String time = req[1];
+            Task temp = new Event(desc, time);
+            tasks.add(temp);
+            storage.writeTask(temp);
+            ui.toScreen("Ok! A new task has been added:",
+                    "    " + temp.toString(),
+                    "You now have " + tasks.size() + " task(s) in total.");
+        } catch (DukeException de) {
+            ui.toScreen("Sorry, the Duke.Event task could not be added.",
+                    "Please include the description and time of this task with /at.");
+        }
+    }
+
+    public void findTask(String key) {
+        if (key.isEmpty()) {
+            ui.toScreen("Please include a search word.");
+            return;
+        }
+
+        String line = "____________________________________________________________";
```

Comment:
Perhaps you could declare line an indent as constants as they are used in more than one function?